test/controllers/unattended_controller_test.rb

```diff
@@ -551,4 +551,114 @@ class UnattendedControllerTest < ActionController::TestCase
       assert_includes @response.body, "#{@host_with_template_subnet.subnet.template.url}/unattended/finish?token=aaaaae"
     end
   end
+
+  context 'ipxe provisioning' do
+    let(:tax_organization) do
+      FactoryBot.create(
+        :organization,
+        ignore_types: ['ProvisioningTemplate']
+      )
+    end
+    let(:tax_location) do
+      FactoryBot.create(
+        :location,
+        ignore_types: ['ProvisioningTemplate']
+      )
+    end
+    let(:ipxe_kind) { TemplateKind.find_by(name: 'iPXE') }
+    let(:ipxe_template) do
+      FactoryBot.create(
+        :provisioning_template,
+        template_kind: ipxe_kind,
+        name: 'iPXE test template',
+        template: "#!ipxe\necho Test build\nexit",
+        organizations: [tax_organization],
+        locations: [tax_location]
+      )
+    end
+    let(:operatingsystem) do
+      FactoryBot.create(
+        :operatingsystem,
+        :with_associations,
+        :with_os_defaults,
+        family: 'Redhat',
+        provisioning_templates: [ipxe_template]
+      )
+    end
+    let(:host) do
+      FactoryBot.create(
+        :host,
+        :managed,
+        operatingsystem: operatingsystem,
+        organization: tax_organization,
+        location: tax_location
+      )
+    end
+
+    setup do
+      disable_orchestration
+      operatingsystem.provisioning_templates << ipxe_template
+    end
+
+    context 'without a host' do
+      test 'should render iPXE error when global iPXE template is not found' do
+        get :host_template, params: { kind: 'iPXE' }, session: set_session_user
+        assert_response :not_found
+        assert_includes @response.body, "Global iPXE template 'iPXE global default' not found"
```

Comment:
Nitpick - do we want this error message to be presented in ipxe syntax like "echo ..."?